controller/contour/deployment.go

```diff
@@ -70,25 +70,30 @@ const (
 )
 
 // ensureDeployment ensures a deployment exists for the given contour.
-func (r *Reconciler) ensureDeployment(ctx context.Context, contour *operatorv1alpha1.Contour) error {
+func (r *Reconciler) ensureDeployment(ctx context.Context, contour *operatorv1alpha1.Contour) (*appsv1.Deployment, error) {
 	desired, err := DesiredDeployment(contour, r.Config.ContourImage)
 	if err != nil {
-		return fmt.Errorf("failed to build deployment: %w", err)
+		return nil, fmt.Errorf("failed to build deployment: %w", err)
 	}
 
 	current, err := r.currentDeployment(ctx, contour)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			return r.createDeployment(ctx, desired)
+			updated, err := r.createDeployment(ctx, desired)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create deployment %s/%s: %w", desired.Namespace, desired.Name, err)
+			}
+			return updated, nil
 		}
-		return fmt.Errorf("failed to get deployment %s/%s: %w", desired.Namespace, desired.Name, err)
+		return nil, fmt.Errorf("failed to get deployment %s/%s: %w", desired.Namespace, desired.Name, err)
 	}
 
-	if err := r.updateDeploymentIfNeeded(ctx, current, desired); err != nil {
-		return fmt.Errorf("failed to update deployment %s/%s: %w", desired.Namespace, desired.Name, err)
+	updated, err := r.updateDeploymentIfNeeded(ctx, current, desired)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update deployment %s/%s: %w", desired.Namespace, desired.Name, err)
```

Comment:
Same comment for `ensureDaemonset`. Should `ensureDeployment` call `currentDeployment` after performing an update?

Reply:
Same feedback as https://github.com/projectcontour/contour-operator/pull/91/files#r518240871.